src/csharp/ChatClient.cs

```diff
@@ -0,0 +1,206 @@
+﻿using Microsoft.Extensions.AI;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.ML.OnnxRuntimeGenAI;
+
+/// <summary>An <see cref="IChatClient"/> implementation based on ONNX Runtime GenAI.</summary>
+public sealed class ChatClient : IChatClient, IDisposable
```

Comment:
Any reasons as to why this class would be sealed? Native AOT? It would be nice to be able to make the CreatePrompt method virtual, or generalizing it somehow.

Reply:
> Any reasons as to why this class would be sealed?

I generally opt to until we have good reason to unseal it, especially in a case like this where interfaces are being implemented, as you then have to design the whole type for the fact that it might be extended.

If there are good reasons to unseal it and virtuals that would be warranted, then it can be.